### PR TITLE
Restrict Graph Editor Zoom

### PR DESCRIPTION
**Pull request by vitvakatu**
### Pull Request Description

[Task link](#181181203).

This is a reincarnation of PR [3273](https://github.com/enso-org/enso/pull/3273). 

The maximum zoom factor of Graph Editor is limited to 1.0x. It is not possible to zoom in from the default camera position. 
Debug Mode (activated with `ctrl-shift-d` shortcut) allows to zoom up to 100.0x (the previous behavior of Graph Editor).

If you enable Debug Mode, then zoom in and disable Debug Mode - you won't see the immediate change of zoom factor back to 1.0x. But it will "jump" (with animation) back once you make a zoom in/out event with your controls.

Video:

https://user-images.githubusercontent.com/6566674/154037310-1d166737-353e-4ae6-aca1-f7840571ab16.mp4

### Important Notes

This is a reincarnation of PR [3273](https://github.com/enso-org/enso/pull/3273). There are two changes since that PR:
1. Fixed bug with GeoMap zooming described [here](https://github.com/enso-org/enso/pull/3290). This is done by restricting `ZoomEvent` API so that it will never contain `amount` which is equal to `0.0`.
2. A few refactoring changes from https://github.com/enso-org/enso/pull/3289 to simplify code a bit.

### Checklist

Please include the following checklist in your PR:

- [x] The documentation has been updated if necessary.
- [x] All code conforms to the [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md), [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md), and [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guides.
- All code has been tested:
  - [x] Unit tests have been written where possible.
  - [x] If GUI codebase was changed: Enso GUI was tested when built using BOTH `./run dist` and `./run watch`.

### CODE REVIEW

File: lib/rust/ensogl/core/src/display/navigation/navigator/events.rs [hunk 1]
```diff
@@ -23,9 +23,15 @@ pub struct ZoomEvent {
 }
 
 impl ZoomEvent {
-    fn new(focus: Vector2<f32>, amount: f32, zoom_speed: f32) -> Self {
+    /// Constructor. Returns `None` if `amount` or `zoom_speed` are equal to zero.
```
Comment (farmaazon) [inline lib/rust/ensogl/core/src/display/navigation/navigator/events.rs]: I'd be more specific here: Not the `zoom_speed` is _almost_ equal (in the range of `f32::Epsilon`).

File: lib/rust/ensogl/core/src/display/navigation/navigator/events.rs [hunk 2]
```diff
@@ -326,8 +333,10 @@ impl NavigatorEvents {
                         MovementType::Zoom { focus } => {
                             let zoom_speed = data.zoom_speed();
                             let zoom_amount = movement_to_zoom(movement);
-                            let zoom_event = ZoomEvent::new(focus, zoom_amount, zoom_speed);
-                            data.on_zoom(zoom_event);
+                            if let Some(zoom_event) = ZoomEvent::new(focus, zoom_amount, zoom_speed)
+                            {
```
Comment (farmaazon) [inline lib/rust/ensogl/core/src/display/navigation/navigator/events.rs]:
```suggestion
                            if let Some(zoom_event) = ZoomEvent::new(focus, zoom_amount, zoom_speed) {
```

**Review by farmaazon — COMMENTED** [review]
Some comments, nothing serious. Now I'm testing it, and will accept if I find nothing.

File: lib/rust/ensogl/core/src/display/navigation/navigator/events.rs (hunk 2 above)
Comment (vitvakatu) [reply lib/rust/ensogl/core/src/display/navigation/navigator/events.rs]: This is `rustfmt`, not me! Looks weird indeed, I changed variable name from `zoom_event` to `event` to fix it.

File: lib/rust/ensogl/core/src/display/navigation/navigator/events.rs (hunk 1 above)
Comment (vitvakatu) [reply lib/rust/ensogl/core/src/display/navigation/navigator/events.rs]: Fixed

**Review by farmaazon — APPROVED** [review]

**Pull request merged**